main.go

```diff
@@ -1,5 +1,94 @@
 package main
 
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/forman"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/reconciler"
+
+	"github.com/jackc/pgconn"
+	"go.uber.org/zap"
+)
 
 func main() {
+	var logger *zap.Logger
+	var err error
+	if os.Getenv("DEVELOPMENT") != "" {
+		logger, err = zap.NewDevelopment()
+	} else {
+		logger, err = zap.NewProduction()
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+	ctx := context.Background()
+	// parse configuration from yaml file
+	conf, err := db.GetConfig(logger, "config/config.yaml")
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	// Establish a connection to the database
+	config, err := pgconn.ParseConfig(conf.GetConnectionString())
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	dbInstance, err := db.NewPGInstance(ctx, logger, *conf)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+	err = dbInstance.Ping(ctx)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	reqChan := forman.GoForman(logger, dbInstance, conf.Workers, reconciler.Reconcile)
+
+	config.OnNotification = func(c *pgconn.PgConn, n *pgconn.Notification) {
+		logger.Info("Received notification", zap.String("payload", n.Payload))
+		req := forman.Request{}
+		err := json.Unmarshal([]byte(n.Payload), &req)
+		if err != nil {
+			logger.Error(err.Error(), zap.String("payload", n.Payload))
+			return
+		}
+		reqChan <- req
+	}
+
+	pgConn, err := pgconn.ConnectConfig(ctx, config)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+	defer pgConn.Close(ctx)
+
+	// Listen for notifications
+	_, err = pgConn.Exec(ctx, "LISTEN provider_cluster;").ReadAll()
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	providers, err := dbInstance.ListProviderClusters(ctx)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	go func(providers []*db.ProviderCluster) {
+		for _, p := range providers {
+			req := forman.Request{}
+			req.Name = p.ClusterID
+			reqChan <- req
+		}
+	}(providers)
```

Comment:
This will queue a reconcile request for every provider, and all at the same time.
We should only queue for providers who had an unprocessed state change or at least not in a stable state

Reply:
Agree, will take care of it in follow up as need to experiment with json, will add a check for hash to compare current spec and desired spec